tests/cases/compiler/correlatedUnions.ts

```diff
@@ -301,3 +301,35 @@ function getValueConcrete<K extends keyof Foo1>(
 ): Foo1[K] | undefined {
   return o[k];
 }
+
+// repro from https://github.com/microsoft/TypeScript/issues/54680

```

Comment:
I'm not 100% sure if I added this test to a proper file

Reply:
Honestly I would just add it to a new file... I wish we just named test cases `issue54680` instead of attempting to throw similar cases in the same file and/or had to come up with some readable name for every esoteric bug 🙁